_posts/2021-04-14-accessing-the-relay-store.markdown

```diff
@@ -0,0 +1,260 @@
+          ConnectionHandler.deleteNode(connection, input.artworkGlobalId)
+        }
+      }
+    },
+    onCompleted: () => actions.deleteArtworkComplete(),
+    onError: actions.deleteArtworkError,
+  })
+```
+
+In my use case, I was not using a Relay mutation because I did not need to modify anything on the server. Since
+Redis is keeping track of our `excludeIDs` for us, any round trip to the server will be up-to-date. We just need to
+modify our local data store.
+
+Relay provides a [separate API method to make local updates][relay-documentation-local-data-updates] to the Relay
+store: `commitLocalUpdate`. `commitLocalUpdate` takes two arguments: the first is the relay environment, which you
+can easily access from the parent relay fragment or refetch container. The second is an updater that takes in the
+store as the first argument. We now have access to the store!
+
+## Finishing the Checklist Task with ConnectionHandler
+
+The main hurdle here was finding an appropriate way to hook into the store for our specific use case——when we do
+not require an update to server data.
+
+But to close us out: Let's delete the item from the connection in the store.
+
+When an item is snoozed, we call `commitLocalUpdate`, pass in the relay environment, and then pass in the updater
+function. Once we have access to the store, our goal is to delete this particular item from the
+`artworksConnection`, which is the GraphQL object returned by our original Relay query.
+
+Because we are dealing with connections, we want to use the [ConnectionHandler
+API][relay-documentation-connection-handler] provided by Relay. `ConnectionHandler.getConnection` takes in the
+connection's parent record (which we can find using the GraphQL ID added as a field on our query for the
+connection) as the first argument and the connection key which can be provided through [Relay’s @connection
+directive][relay-modern-connection-derivative].
+
+Once we have the connection, we will use `ConnectionHandler.deleteNode` which takes the connection as the first
+argument and the id to be deleted, which we can also easily access using the GraphQL ID added as a field to the
+query for the item.
+
+Bonus: Because `commitLocalUpdate` works anywhere in Relay land, we got to perform this deletion exactly where the
+"snooze" action is happening: in the child item component. (In our previous solution, we had to manage the state of
+the children from their parent component, which wasn't as intuitive.)
+
+```js
+commitLocalUpdate(relay.environment, (store) => {
+  const parentRecord = store.get(parentID)
+
+  if (parentRecord) {
+    const artworksConnection = ConnectionHandler.getConnection(
+      parentRecord,
+      "ArtworksMissingMetadata_partner_artworksConnection"
+    )
+    if (artworksConnection) {
+      ConnectionHandler.deleteNode(artworksConnection, id)
+    }
+  }
+})
+```
+
+## Key Takeaways
+
+1. Relay is great because it couples our data fetch with our React component.
```

Comment:
We should align any changes in the beginning of the article with a change here, to indicate it's not the _fetch_ that's coupled so much as the _data requirements_.